packages/flagship/src/lib/ios.ts

```diff
@@ -259,13 +259,42 @@ export function usageDescription(configuration: Config): void {
         infoPlist,
         '<key>UIRequiredDeviceCapabilities</key>',
         `<key>${usage.key}</key><string>${
-          usage.string
+        usage.string
         }</string><key>UIRequiredDeviceCapabilities</key>`
       );
     }
   });
 }
 
+/**
+ * Adds usage description from the project configuration.
+ *
+ * @param {object} configuration The project configuration.
+ */
+export function backgroundModes(configuration: Config): void {
+  if (!configuration.UIBackgroundModes) {
+    return;
+  }
+
+  helpers.logInfo(
+    `updating iOS background modes: [${configuration.UIBackgroundModes.map(u => u.string)}]`
+  );
+
+  const infoPlist = path.ios.infoPlistPath(configuration);
+
+  fs.update(
+    infoPlist,
+    '<key>UIRequiredDeviceCapabilities</key>',
+    `<key>UIBackgroundModes</key>
+      <array>
+      ${configuration.UIBackgroundModes.forEach(mode => {
```

Comment:
The `forEach` here needs to be `map`. As is, this just puts "undefined" in the file.

Reply:
@Cauldrath you're right, PR updated.